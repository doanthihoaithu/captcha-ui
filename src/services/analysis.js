import { requestServices } from 'services';
import * as constants from 'constants/index'


const analyzeQuery = (accessToken, data) => requestServices.customAxios.post(
    constants.analysis_analyze_query,
    data,
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data'
        }
    }
)

const generateQuery = (data) => requestServices.customAxios.post(
    constants.generate_captcha_query,
    data,
    {
        headers: {
            // 'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data',
            // 'Content-Type': 'application/json'
        }
    }
)

export default {
    analyzeQuery,
    generateQuery
}