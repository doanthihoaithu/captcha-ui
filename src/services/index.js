// @flow
export { default as userServices } from './users';
export { default as requestServices } from './requests';
export { default as rootReducer } from './rootReducer';
export { default as store } from './store'
