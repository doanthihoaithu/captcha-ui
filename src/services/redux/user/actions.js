import { LOG_IN, LOG_OUT, UPDATE_AVATAR } from './actionTypes';

export const logIn = (data) => ({
	type: LOG_IN,
	username: data.username,
	access_token: data.access_token,
	refresh_token: data.refresh_token
});

export const logOut = () => ({
	type: LOG_OUT,
	username: undefined,
	access_token: undefined,
	refresh_token: undefined,
});