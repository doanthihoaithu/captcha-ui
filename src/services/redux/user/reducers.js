import Cookies from 'universal-cookie'
import { LOG_IN, LOG_OUT, UPDATE_AVATAR } from './actionTypes';

const cookies = new Cookies();

const initialState = {
	isAuthenticated: !!cookies.get("leave")
}

export default (state = initialState, action) => {
	switch (action.type) {
		case LOG_IN:
			return {
				...state,
				isAuthenticated: true,
				username: action.username,
				accessToken: action.access_token,
				refreshToken: action.refresh_token,
			};
		case LOG_OUT:
			cookies.remove('leave');
			return {
				...state,
				isAuthenticated: false,
				username: undefined,
				accessToken: undefined,
				refreshToken: undefined,
			};
		default:
			return state;
	}
}