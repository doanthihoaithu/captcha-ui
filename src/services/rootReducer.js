import { combineReducers } from 'redux';
import userReducer from './redux/user/reducers';
import statusReducer from './redux/status/reducers';

export default combineReducers({
	user: userReducer,
	status: statusReducer
});