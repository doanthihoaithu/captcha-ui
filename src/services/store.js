import { createStore } from 'redux';
import { rootReducer } from './../services';

const store = createStore(rootReducer);

export default store;
