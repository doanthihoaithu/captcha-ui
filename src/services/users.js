// @flow
import { requestServices } from 'services';
import * as constants from 'constants/index'

// const fetchUsers = () => requestServices.customAxios.get('/').then(res => res.data);

const signUserIn = (data) => requestServices.customAxios.post(
    constants.user_authenticate_address,
    data,
    {
        header: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*',
        }
    }
)

const signUserOut = (accessToken) => requestServices.customAxios.post(
    constants.user_sign_out_address,
    {},
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const registerNewUser = (data) => requestServices.customAxios.post(
    constants.user_sign_up_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const resetPassword = (data) => requestServices.customAxios.post(
    constants.user_forget_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const changePassword = (data, accessToken) => requestServices.customAxios.post(
    constants.user_change_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${accessToken}`
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const updateAvatar = (username, data, accessToken) => requestServices.customAxios.post(
    constants.user_update_avatar_address + username,
    data,
    {
        headers: {
            'Content-Type': 'multipart/form-data',
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const getAvatar = (username) => requestServices.customAxios.get(
    constants.user_get_avatar_address + username,
)

export default {
    signUserIn,
    signUserOut,
    registerNewUser,
    resetPassword,
    changePassword,
    updateAvatar,
    getAvatar,
};
