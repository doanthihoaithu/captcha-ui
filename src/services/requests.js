// @flow
import axios from 'axios';
import * as actions from 'services/redux/user/actions';
import { store } from 'services'
import { toast } from 'react-toastify'

export const BASE_API_URL: string = `${process.env.REACT_APP_BASE_API_URL || 'localhost:3000'}`;

const customAxios = axios.create({
  baseURL: BASE_API_URL,
});

customAxios.interceptors.response.use(
  response => response,
  error => {
    // Add logic to handle error here
    // console.log(error.response.data)
    if (error.response && error.response.data.code === 500) {
      switch (error.response.data.message) {
        case "Signature has expired":
          store.dispatch(actions.logOut());
          toast.error('Your session has expired. Please sign in to continue');
          return Promise.reject(error);
        default:
          return Promise.reject(error);
      }
    }

    return Promise.reject(error);
  },
);

export default { customAxios };
