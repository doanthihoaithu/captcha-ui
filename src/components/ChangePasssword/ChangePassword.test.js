import React from 'react'
import { shallow, render, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { store, userServices } from './../../services'
import * as actions from './../../services/redux/user/actions';
import { BrowserRouter as Router } from 'react-router-dom';
import { act } from 'react-dom/test-utils';

import ChangePassword from '.'

let userData = {
    username: 'admintest',
    access_token: 'access_token',
    refresh_token: 'access_token',
}

//create user data in redux store
store.dispatch(actions.logIn(userData))

let component;
let changePasswordAPI;

beforeEach(() => {
    component = mount(
        <Router>
            <Provider store={store}>
                <ChangePassword />
            </Provider>
        </Router>
    )

    changePasswordAPI = jest.spyOn(userServices, 'changePassword').mockReturnValue(Promise.resolve("value"));
})

afterEach(() => {
    changePasswordAPI.mockReset();
    component.unmount();
})

it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
})


it('should not call function changePassword when submit nothing', () => {
    let changePasswordForm = component.find("#changePasswordForm");

    changePasswordForm.simulate('submit');

    expect(changePasswordAPI).toHaveBeenCalledTimes(0);
})

it('should call function changePassword when submit valid data', () => {
    let changePasswordForm = component.find("#changePasswordForm");

    component.find("input[id='oldPassword']").getDOMNode().value = ("oldpassword")
    component.find("input[id='newPassword']").getDOMNode().value = ("newpassword")
    component.find("input[id='confirmNewPassword']").getDOMNode().value = ("newpassword")

    component.update()

    expect(component.find("input")).toHaveLength(3)
    changePasswordForm.simulate('submit');

    expect(changePasswordAPI).toHaveBeenCalledTimes(1);
})

it('should not call function changePassword when submit invalid data', () => {
    let changePasswordForm = component.find("#changePasswordForm");

    component.find("input[id='oldPassword']").getDOMNode().value = ("")
    component.find("input[id='newPassword']").getDOMNode().value = ("newpass")
    component.find("input[id='confirmNewPassword']").getDOMNode().value = ("newpass")

    component.update()

    expect(component.find("input")).toHaveLength(3)
    changePasswordForm.simulate('submit');

    expect(changePasswordAPI).toHaveBeenCalledTimes(0);
})