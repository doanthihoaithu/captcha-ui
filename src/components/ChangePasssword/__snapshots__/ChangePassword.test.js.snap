// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <ChangePassword>
        <div
          className="container-contact100 animated fadeIn fast "
        >
          <div
            className="wrap-contact100"
          >
            <button
              className="btn-hide-contact100"
            >
              <i
                className="zmdi zmdi-close"
              />
            </button>
            <div
              className="contact100-form-title"
              style={
                Object {
                  "backgroundImage": "url(bg-02.jpg)",
                }
              }
            >
              <span>
                Change Password
              </span>
            </div>
            <form
              className="contact100-form validate-form"
              id="changePasswordForm"
              onSubmit={[Function]}
            >
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  aria-describedby="oldPasswordHelp"
                  className="input100"
                  id="oldPassword"
                  pattern="^(?=.{8,150}$)[a-zA-Z0-9]+$"
                  placeholder="Enter old password"
                  required={true}
                  type="password"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="oldPassword"
                >
                  <span
                    className="lnr lnr-lock m-b-2"
                  />
                </label>
              </div>
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  className="input100"
                  id="newPassword"
                  pattern="^(?=.{8,150}$)[a-zA-Z0-9]+$"
                  placeholder="Enter new password"
                  required={true}
                  type="password"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="newPassword"
                >
                  <span
                    className="lnr lnr-cog m-b-5"
                  />
                </label>
              </div>
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  className="input100"
                  id="confirmNewPassword"
                  pattern="^(?=.{8,150}$)[a-zA-Z0-9]+$"
                  placeholder="Confirm new password"
                  required={true}
                  type="password"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="newPasswordConfirm"
                >
                  <span
                    className="lnr lnr-redo m-b-5"
                  />
                </label>
              </div>
              <div
                className="w-100 text-center mb-1"
              >
                <span
                  className="text-danger"
                  hidden={true}
                >
                  Password not matched. Please try again!
                </span>
              </div>
              <div
                className="w-100 text-center mb-1"
              >
                <span
                  className="text-danger"
                  hidden={true}
                >
                  Something went wrong. Please try again!
                </span>
              </div>
              <div
                className="container-contact100-form-btn"
              >
                <button
                  className="contact100-form-btn"
                  type="submit"
                >
                  Reset Password
                </button>
              </div>
            </form>
            <div
              className="d-flex flex-column justify-content-center align-items-center mb-4"
            >
              <Link
                className="text-info"
                to="/"
              >
                <a
                  className="text-info"
                  href="/"
                  onClick={[Function]}
                >
                  Back to Home
                </a>
              </Link>
            </div>
          </div>
        </div>
      </ChangePassword>
    </Provider>
  </Router>
</BrowserRouter>
`;
